psa-crypto/src/operations/key_derivation.rs

```diff
@@ -0,0 +1,92 @@
+// Copyright 2020 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+
+//! # Key Derivation operations
+
+use super::key_management::complete_new_key_operation;
+use crate::initialized;
+use crate::types::key::Attributes;
+use crate::types::key::Id;
+use crate::types::key_derivation::Operation;
+use crate::types::status::{Result, Status};
+use core::convert::{TryFrom, TryInto};
+
+/// This function calculates output bytes from a key derivation algorithm and uses those bytes to generate a key deterministically.
+/// The key's location, usage policy, type and size are taken from attributes.
+/// # Example
+///
+/// ```
+/// use psa_crypto::operations::{key_derivation, key_management};
+/// use psa_crypto::types::key::{Attributes, Type, Lifetime, Policy, UsageFlags};
+/// use psa_crypto::types::algorithm::{Hash, KeyDerivation};
+/// use psa_crypto::types::key_derivation::{Operation, Inputs, Input};
+///
+/// # const KEY_DATA: [u8; 23] = [0; 23];
+/// # let mut attributes = Attributes {
+/// #     key_type: Type::Derive,
+/// #     bits: 0,
+/// #     lifetime: Lifetime::Volatile,
+/// #     policy: Policy {
+/// #         usage_flags: UsageFlags {
+/// #             derive: true,
+/// #             ..Default::default()
+/// #         },
+/// #         permitted_algorithms: KeyDerivation::Hkdf {
+/// #            hash_alg: Hash::Sha256,
+/// #         }.into()
+/// #     }
+/// # };
+///
+/// # let mut derived_key_attributes = Attributes {
```

Comment:
Can the `key_type` and the `permitted_algorithms` of the derived key can be anything? Could it be used for RSA signing for example?

Reply:
I believe so. Looking at the [description for `psa_key_derivation_output_key`](https://armmbed.github.io/mbed-crypto/html/api/ops/kdf.html#c.psa_key_derivation_output_key), there doesn't seem to be a restriction on the type of key. At the end it says 'For other key types', so presumably allows all types.